### DISCUSSION TITLE
How to Get The World Position From The Cursor ( Screen ) Space?

### DISCUSSION THREAD

**Discussion by zicklag** [bevyengine/bevy#6125] [General]
This is something I've run into needing to do a few times and didn't quite figure it out, but I just found this example in `bevy_ecs_tilemap` that shows how simple it really is!

https://github.com/StarArawn/bevy_ecs_tilemap/blob/de8d005331dde7d9ebb8c4fe1b36769b84e44be5/examples/mouse_to_tile.rs#L291-L306

```rust
pub fn cursor_pos_in_world(
    windows: &Windows,
    cursor_pos: Vec2,
    cam_t: &Transform,
    cam: &Camera,
) -> Vec3 {
    let window = windows.primary();

    let window_size = Vec2::new(window.width(), window.height());

    // Convert screen position [0..resolution] to ndc [-1..1]
    // (ndc = normalized device coordinates)
    let ndc_to_world = cam_t.compute_matrix() * cam.projection_matrix().inverse();
    let ndc = (cursor_pos / window_size) * 2.0 - Vec2::ONE;
    ndc_to_world.project_point3(ndc.extend(0.0))
}
```

Sharing here in case anybody else comes searching for it like I did. It seems like this would also be a good utility function to include on the camera, similar to the function that lets you get the screen space from the world space.

**Comment by tim-blackbird** [comment]
I happened to be working on this today. The PR is now up :) #6126

**Reply by zicklag** [reply]
Wow, what a coincidence!

Thanks for that!
